driver-core/src/main/com/mongodb/ReadPreference.java

```diff
@@ -599,6 +652,26 @@ private static TaggableReadPreference valueOf(final String name, final List<TagS
         private PrimaryReadPreference() {
         }
 
+        @Override
+        public ReadPreference withTagSet(final TagSet tagSet) {
+            throw new IllegalArgumentException("Primary read preference can not also specify tag sets");
```

Comment:
`UnsupportedOperationException` is the one to throw for these methods.

Reply:
Changed.